pydynamo_brain/pydynamo_brain/ui/radiiPainter.py

```diff
@@ -83,63 +84,98 @@ def drawPoint(self, point, selectedPointID):
             )
             self.maybeDrawText(x, y, point)
 
-    def returnRadiusCoord(self, point, radius):
+    def returnRadiusCoord(self, point, radius, realRadius):
         #point radius will be drawn from
         delta = 1
-
-        """        #previous point
-        posX, posY, negX, negY = x2, y2, x2, y2,"""
+        #Drawing the radius calipers for root points
         if  point.isRoot():
-            #print(point.isRoot)
             nextPoint  = point.nextPointInBranch(delta)
-            #print(nextPoint)
             p2Loc = point.location
-            #print(p2Loc)
-            x2, y2, z2 = self.zoomedLocation(p2Loc)
 
+            x2, y2, z2 = self.zoomedLocation(p2Loc)
             p1Loc = nextPoint.location
             x1, y1, z1 = self.zoomedLocation(p1Loc)
 
-
-        else:
+            #length between two points
+            L = math.sqrt(math.pow((x2-x1),2)+math.pow((y2-y1),2))
+            if L == 0:
+                L = 1
+            #all solutions for the plotting radius
+            #assuming a right triangle
+            negX = x2 -((radius*(y1-y2)/L))
+            posX = x2 +((radius*(y1-y2)/L))
+            posY = y2 +((radius*(x1-x2)/L))
+            negY = y2 -((radius*(x1-x2)/L))
+
+            return negX, posY, posX, negY
+        #drawing the raidus calipers for terminal points
+        elif point.isLastInBranch():
             p2Loc = point.location
             x2, y2, z2 = self.zoomedLocation(p2Loc)
             point1 =  point.pathFromRoot()[-2:][0]
             p1Loc = point1.location
             x1, y1, z1 = self.zoomedLocation(p1Loc)
 
-        #length between two points
-        L = math.sqrt(math.pow((x2-x1),2)+math.pow((y2-y1),2))
-        if L == 0:
-            L = 1
-        #all solutions for the plotting radius
-        #assuming a right triangle
-        negX = x2 -((radius*(y1-y2)/L))
-        posX = x2 +((radius*(y1-y2)/L))
-        posY = y2 +((radius*(x1-x2)/L))
-        negY = y2 -((radius*(x1-x2)/L))
+            #length between two points
+            L = math.sqrt(math.pow((x2-x1),2)+math.pow((y2-y1),2))
+            if L == 0:
+                L = 1
+            #all solutions for the plotting radius
+            #assuming a right triangle
+            negX = x2 -((radius*(y1-y2)/L))
+            posX = x2 +((radius*(y1-y2)/L))
+            posY = y2 +((radius*(x1-x2)/L))
+            negY = y2 -((radius*(x1-x2)/L))
+            return negX, posY, posX, negY
+        #drawing calipers for mid branch points, angles based off an average of
+        #previous and last point
+        else:
+            #previous point
+            pointCoord = point.location
+            xPix, yPix, zPix = self.zoomedLocation(pointCoord)
+            xyPix = np.array([xPix, yPix])
+
+            previousPoint =  point.pathFromRoot()[-2]
+            nextPoint  = point.nextPointInBranch(delta)
+
+
+            PA = self.caliperPoints(point, nextPoint, ifNext= True)
```

Comment:
I'd recommend not using all-caps variables here - that signifies global constants.
pA / pB / pC is fine here.